src/server/initServer.ts

```diff
@@ -267,13 +263,144 @@ const initServer = (
+          query.errorCode = { $regex: actionErrorFilterState.errorCode, $options: 'i' };
+        }
+
+        if (actionErrorFilterState.type === 'action') {
+          const selectedTargets = Object.keys(actionErrorFilterState.target).filter(
+            (target) => { return actionErrorFilterState.target[target]; },
+          );
+          const selectedActions = Object.keys(actionErrorFilterState.action).filter(
+            (action) => { return actionErrorFilterState.action[action]; },
+          );
+          if (selectedTargets.length > 0) {
+            query.target = { $in: selectedTargets };
+          }
+          if (selectedActions.length > 0) {
+            query.action = { $in: selectedActions };
+          }
+        }
+
+        /* ------------ Advanced Filter ----------- */
+        const { advancedFilterState } = filters;
+
+        if (advancedFilterState.userFirstName) {
+          query.userFirstName = { $regex: advancedFilterState.userFirstName, $options: 'i' };
+        }
+
+        if (advancedFilterState.userLastName) {
+          query.userLastName = { $regex: advancedFilterState.userLastName, $options: 'i' };
+        }
+
+        if (advancedFilterState.userEmail) {
+          query.userEmail = { $regex: advancedFilterState.userEmail, $options: 'i' };
+        }
+
+        if (advancedFilterState.userId) {
+          query.userId = parseInt(advancedFilterState.userId, 10);
+        }
+
+        const roles = [];
+        if (advancedFilterState.includeLearners) {
+          roles.push({ isLearner: true });
+        }
+        if (advancedFilterState.includeTTMs) {
+          roles.push({ isTTM: true });
+        }
+        if (advancedFilterState.includeAdmins) {
+          roles.push({ isAdmin: true });
+        }
+        if (roles.length > 0) {
+          query.$and = [{ $or: roles }];
+        }
+
+        if (advancedFilterState.courseId) {
+          query.courseId = parseInt(advancedFilterState.courseId, 10);
+        }
+
+        if (advancedFilterState.courseName) {
+          query.courseName = { $regex: advancedFilterState.courseName, $options: 'i' };
+        }
+
+        if (advancedFilterState.isMobile !== undefined) {
+          query['device.isMobile'] = advancedFilterState.isMobile;
+        }
+
+        if (advancedFilterState.source) {
+          query.source = advancedFilterState.source;
+        }
+
+        if (advancedFilterState.routePath) {
+          query.routePath = { $regex: advancedFilterState.routePath, $options: 'i' };
+        }
+
+        if (advancedFilterState.routeTemplate) {
+          query.routeTemplate = { $regex: advancedFilterState.routeTemplate, $options: 'i' };
+        }
+
         // Query for logs
         const response = await _logCollection.findPaged({
-          query: {
-            year,
-            month,
-          },
-          perPage: 1000,
+          query,
+          perPage: 50,
```

Comment:
Generally, needs a lot of comments because this code is super complicated. Good work! I just want others to be able to understand it